ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/PeriodicRecovery.java

```diff
@@ -275,7 +275,7 @@ public Mode suspendScan (boolean async)
                setMode(Mode.SUSPENDED);
                _stateLock.notifyAll();
            }
-           if (!async) {
+           if (!recoveryPropertyManager.getRecoveryEnvironmentBean().isWaitForFinalRecovery() && !async) {
```

Comment:
@jmfinelli please can you confirm what the impact of adding this additional possibility to not wait at _stateLock.wait() will be in suspendScan - it would already only wait if SCANNING, is it possible for another scan to start here?

Reply:
IIRC, what I wanted to achieve here with this change was to skip waiting for the last scan (i.e. `async false if the calling thread should wait for any in-progress scan to complete before returning`). The reason is: if  `isWaitForFinalRecovery()` returns true, `PeriodicRecovery` would have already terminated all possible scans and `currentMode` should be `SUSPENDED`. Having said that, to answer your question, I don't think another scan can start here. Nevertheless, I think that skipping this `if` statement based on `isWaitForFinalRecovery()` is an improvement that shouldn't have negative consequences (i.e. bugs).

Reply:
On the other hand, if you want to avoid modifying this `if` statement, we could just add a comment to explain why `isWaitForFinalRecovery() == true` means that `while (getStatus() == Status.SCANNING)` will be skipped.

Reply:
I have noticed something, operations in the environment beans are not intended to be altered at runtime like I am thinking this setWaitForFinalRecovery might be intended to be used. We will probably want to move that out of the environment bean unless it is intended to be set once at boot up and not changed (not what I am expecting).

Reply:
Yeah, I see your point. +1

Reply:
I have made a local change to address the point about where the method is and to add a comment about isWaitForFinalRecovery